--- allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
+++ allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
@@ -216,6 +216,10 @@
     std::vector<allocator_test_utils::block_info> info;
     buddy_iterator end_it = end();
     for (buddy_iterator it = begin(); it != end_it; ++it) {
+        if (!it.operator*()) {
+            break;
+        }
+        if (it.operator*() >= end_it.operator*()) break;
         allocator_test_utils::block_info block;
         block.block_size = it.size();
         block.is_block_occupied = it.occupied();
