--- allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
+++ allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
@@ -107,10 +107,11 @@
     }

     best_block->occupied = true;
-    if (best_block->size >= min_k) {
-        *reinterpret_cast<void**>(reinterpret_cast<char*>(best_block) + sizeof(block_metadata)) = _trusted_memory;
-    }
-    return reinterpret_cast<char*>(best_block) + occupied_block_metadata_size;
+    if (best_block->size >= min_k) {
+        *reinterpret_cast<void**>(reinterpret_cast<char*>(best_block) + sizeof(block_metadata)) = _trusted_memory;
+        return reinterpret_cast<char*>(best_block) + occupied_block_metadata_size;
+    }
+    return reinterpret_cast<char*>(best_block) + free_block_metadata_size;
 }
