--- allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
+++ allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
@@ -165,52 +165,65 @@

 allocator_buddies_system::buddy_iterator allocator_buddies_system::begin() const noexcept
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator allocator_buddies_system::begin() const noexcept", "your code should be here...");
+    char* ptr = reinterpret_cast<char*>(_trusted_memory);
+    ptr += sizeof(allocator_dbg_helper*) + sizeof(fit_mode) + sizeof(unsigned char) + sizeof(std::mutex) + sizeof(std::pmr::memory_resource*);
+    return buddy_iterator(ptr);
 }

 allocator_buddies_system::buddy_iterator allocator_buddies_system::end() const noexcept
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator allocator_buddies_system::end() const noexcept", "your code should be here...");
+    char* ptr = reinterpret_cast<char*>(_trusted_memory);
+    unsigned char k_size = *reinterpret_cast<unsigned char*>(ptr + sizeof(allocator_dbg_helper*) + sizeof(fit_mode));
+    return buddy_iterator(ptr + (1ULL << k_size));
 }

 bool allocator_buddies_system::buddy_iterator::operator==(const allocator_buddies_system::buddy_iterator &other) const noexcept
 {
-    throw not_implemented("bool allocator_buddies_system::buddy_iterator::operator==(const allocator_buddies_system::buddy_iterator &) const noexcept", "your code should be here...");
+    return _block == other._block;
 }

 bool allocator_buddies_system::buddy_iterator::operator!=(const allocator_buddies_system::buddy_iterator &other) const noexcept
 {
-    throw not_implemented("bool allocator_buddies_system::buddy_iterator::operator!=(const allocator_buddies_system::buddy_iterator &) const noexcept", "your code should be here...");
+    return _block != other._block;
 }

 allocator_buddies_system::buddy_iterator &allocator_buddies_system::buddy_iterator::operator++() & noexcept
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator &allocator_buddies_system::buddy_iterator::operator++() & noexcept", "your code should be here...");
+    if (_block) {
+        unsigned char k_size = reinterpret_cast<block_metadata*>(_block)->size;
+        _block = reinterpret_cast<char*>(_block) + (1ULL << k_size);
+    }
+    return *this;
 }

 allocator_buddies_system::buddy_iterator allocator_buddies_system::buddy_iterator::operator++(int n)
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator allocator_buddies_system::buddy_iterator::operator++(int)", "your code should be here...");
+    buddy_iterator temp = *this;
+    ++(*this);
+    return temp;
 }

 size_t allocator_buddies_system::buddy_iterator::size() const noexcept
 {
-    throw not_implemented("size_t allocator_buddies_system::buddy_iterator::size() const noexcept", "your code should be here...");
+    return 1ULL << reinterpret_cast<block_metadata*>(_block)->size;
 }

 bool allocator_buddies_system::buddy_iterator::occupied() const noexcept
 {
-    throw not_implemented("bool allocator_buddies_system::buddy_iterator::occupied() const noexcept", "your code should be here...");
+    return reinterpret_cast<block_metadata*>(_block)->occupied;
 }

 void *allocator_buddies_system::buddy_iterator::operator*() const noexcept
 {
-    throw not_implemented("void *allocator_buddies_system::buddy_iterator::operator*() const noexcept", "your code should be here...");
+    return _block;
 }

 allocator_buddies_system::buddy_iterator::buddy_iterator(void *start)
+    : _block(start)
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator::buddy_iterator(void *)", "your code should be here...");
 }

 allocator_buddies_system::buddy_iterator::buddy_iterator()
+    : _block(nullptr)
 {
-    throw not_implemented("allocator_buddies_system::buddy_iterator::buddy_iterator()", "your code should be here...");
 }
