--- allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
+++ allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
@@ -185,22 +185,38 @@

 bool allocator_buddies_system::do_is_equal(const std::pmr::memory_resource &other) const noexcept
 {
-    throw not_implemented("bool allocator_buddies_system::do_is_equal(const std::pmr::memory_resource &other) const noexcept", "your code should be here...");
+    auto* other_buddies = dynamic_cast<const allocator_buddies_system*>(&other);
+    return other_buddies && _trusted_memory == other_buddies->_trusted_memory;
 }

 inline void allocator_buddies_system::set_fit_mode(
     allocator_with_fit_mode::fit_mode mode)
 {
-    throw not_implemented("inline void allocator_buddies_system::set_fit_mode(allocator_with_fit_mode::fit_mode)", "your code should be here...");
+    std::lock_guard<std::mutex> lock(*reinterpret_cast<std::mutex*>(
+        reinterpret_cast<char*>(_trusted_memory) + sizeof(allocator_dbg_helper*) + sizeof(fit_mode) + sizeof(unsigned char)));
+    *reinterpret_cast<fit_mode*>(reinterpret_cast<char*>(_trusted_memory) + sizeof(allocator_dbg_helper*)) = mode;
 }


 std::vector<allocator_test_utils::block_info> allocator_buddies_system::get_blocks_info() const noexcept
 {
-    throw not_implemented("std::vector<allocator_test_utils::block_info> allocator_buddies_system::get_blocks_info() const noexcept", "your code should be here...");
+    std::lock_guard<std::mutex> lock(*reinterpret_cast<std::mutex*>(
+        reinterpret_cast<char*>(_trusted_memory) + sizeof(allocator_dbg_helper*) + sizeof(fit_mode) + sizeof(unsigned char)));
+    return get_blocks_info_inner();
 }

 std::vector<allocator_test_utils::block_info> allocator_buddies_system::get_blocks_info_inner() const
 {
-    throw not_implemented("std::vector<allocator_test_utils::block_info> allocator_buddies_system::get_blocks_info_inner() const", "your code should be here...");
+    std::vector<allocator_test_utils::block_info> info;
+    buddy_iterator end_it = end();
+    for (buddy_iterator it = begin(); it != end_it; ++it) {
+        allocator_test_utils::block_info block;
+        block.block_size = it.size();
+        block.is_block_occupied = it.occupied();
+        info.push_back(block);
+    }
+    return info;
 }
