--- allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
+++ allocator/allocator_buddies_system/src/allocator_buddies_system.cpp
@@ -102,6 +102,9 @@
         unsigned char current_k = reinterpret_cast<block_metadata*>(*it)->size;
         if (current_k >= required_k) {
             if (mode == fit_mode::first_fit) {
                 best_block = reinterpret_cast<block_metadata*>(*it);
                 break;
             } else if (mode == fit_mode::the_best_fit) {
                 if (!best_block || current_k < best_block->size) {
                     best_block = reinterpret_cast<block_metadata*>(*it);
                 }
             } else if (mode == fit_mode::the_worst_fit) {
                 if (!best_block || current_k > best_block->size) {
                     best_block = reinterpret_cast<block_metadata*>(*it);
                 }
             }
         }
     }
